### PR TITLE
matter: upmerge sdk-connectedhomeip to 1.0.0.2

### PR DESCRIPTION
**Pull request by Damian-Nordic**
1. Bump Matter fork to 1.0.0.2 tag.
2. Update ZAP configuration of all Matter apps.
3. Add temporary BYPASS_IDL flag in our CMake files until we add IDL requirements to our CI docker images.

**Reviewer note: files in zap-generated directories are auto-generated and do not need to be reviewed**
test-sdk-nrf: PR-362

### CODE REVIEW

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -90,7 +90,7 @@ See `Matter samples`_ for the list of changes for the Matter samples.
 Matter fork
 +++++++++++
 
-The Matter fork in the |NCS| (``sdk-connectedhomeip``) contains all commits from the upstream Matter repository up to, and including, ``bc6b43882a56ddb3e94d3e64956bd5f3292b4058``.
+The Matter fork in the |NCS| (``sdk-connectedhomeip``) contains all commits from the upstream Matter repository up to, and including, ``1.0.0.2`` tag.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:93]:
```suggestion
The Matter fork in the |NCS| (``sdk-connectedhomeip``) contains all commits from the upstream Matter repository up to, and including, the ``1.0.0.2`` tag.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -90,7 +90,7 @@ See `Matter samples`_ for the list of changes for the Matter samples.
...
 
 The following list summarizes the most important changes inherited from the upstream Matter:
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:95]: Any changes worth of mentioning in the list in line 97?

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst (hunk 2 above)
Comment (kkasperczyk-no) [reply doc/nrf/releases/release-notes-changelog.rst:95]: I think we could mention new log regions filtering and memory optimizations we introduced.

Comment (kkasperczyk-no) [reply doc/nrf/releases/release-notes-changelog.rst:95]: Oh ok, my bad I thought those are not patches anymore, but parts of upmerge.

Comment (greg-fer) [reply doc/nrf/releases/release-notes-changelog.rst:95]: OK, up to you what should be mentioned here, but can we add at least the information that the basic cluster was renamed? :)

Comment (Damian-Nordic) [reply doc/nrf/releases/release-notes-changelog.rst:95]: @kkasperczyk-no 1.0.0.2 contains most changes up to mid-December so not this time :)
@greg-fer Right, already added 3 points, but will also mention about Basic cluster.

**Review by greg-fer — APPROVED** [review]

**Review by ArekBalysNordic — APPROVED** [review]

**Review by kkasperczyk-no — APPROVED** [review]

**Pull request merged**